### DISCUSSION TITLE
Announcement: Updating and Maintaining Foundation

### DISCUSSION THREAD

**Discussion by munawar-js** [stackrole/gatsby-starter-foundation#91] [Ideas]
We updated gatsby-starter-foundation from Gatsby v3 to v4.7 last year, but Gatsby had already moved to v5.8 two years ago.

With [@GatsbyJS](https://x.com/GatsbyJS) now part of [@Netlify](https://x.com/Netlify) and NetlifyCMS rebranded to [@Decap_CMS](https://x.com/Decap_CMS) under PM TechHub.

We are considering to transition our template to a new framework and CMS.

**Comment by munawar-js** [comment]
Things to consider before the transition.

**Tech stack:** 
- Gatsbyjs on React
- NetlifyCMS with Markdown & JSON, queried with Graphql
- Styling with ThemeUI, Sass, Prismjs(syntax highlighter)

**Pages:**
- Home bio, picture, social links and blog posts
- Blog with blog posts
  - Blog post
- About
- Contact via Netlify Email 

**Features:** 
- Theme mode toggle [Dark & Light]
- Accent color
- Manifest.json
- Sitemap
- Blog posts CRUD via git-based NetlifyCMS
- SEO options
- Open graph
- Twitter cards
 
**Integrations:** 
- Google Analytics
- Search [elasticlunr]

**Reply by charles-hood** [reply]
Why move away to something else instead of just updating?

**Reply by munawar-js** [reply]
Great question—I’ve been wrestling with this too. Here’s what I’ve realized during the update process:

  1. **Performance**: Gatsby v4.7 feels sluggish in our current setup. Even if we update to v5.8 (the last major update from two years ago), subsequent changes have mostly been minor bug fixes and maintenance tasks, as seen in [Gatsby’s changelog](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/CHANGELOG.md).
  2. **Stagnation**: 
	  - Gatsby’s development has slowed significantly, and its plugin ecosystem is largely stuck on versions compatible with 4.0.0@next rather than the latest 5.x versions.
	  - Similarly, NetlifyCMS has rebranded to Decap CMS, but the updates have been mostly incremental. You can see the details in [Decap CMS’s changelog](https://github.com/decaporg/decap-cms/blob/main/packages/decap-cms/CHANGELOG.md). The rebrand hasn’t brought significant advancements, raising concerns about its long-term viability for our needs.

Given these factors, I’m uncertain about the future direction for gatsby-starter-foundation, which has led us to consider transitioning to a new framework and CMS.

**Reply by charles-hood** [reply]
Have you looked at TinaCMS?

**Reply by munawar-js** [reply]
Yes, I’m exploring TinaCMS. It’s a promising git-based CMS, and its cloud model with a free tier looks appealing for **Foundation**. The visual editing features are particularly interesting, but I still need to dig deeper to determine if it’s the best fit for us.

Are there any other alternatives you’d recommend considering before we make the switch?

**Reply by munawar-js** [reply]
> Have you looked at TinaCMS?

<img width="592" alt="image" src="https://github.com/user-attachments/assets/89c20636-91d2-495e-aa28-e28c4c268264">
<img width="534" alt="image" src="https://github.com/user-attachments/assets/29fd2d2f-e92f-4eb3-b365-98476ee81809">

TinaCMS is not the right fit for our **Content Modeling**  [NetlifyCMS Config](https://github.com/stackrole/gatsby-starter-foundation/blob/main/static/admin/config.yml)

**Reply by charles-hood** [reply]
Seems like Decap CMS may be the way to go then.

**Reply by munawar-js** [reply]
Yes, Despite lack of updates its pretty solid.

**Comment by munawar-js** [comment]
Our **top priority** is to help users smoothly migrate their content and settings from Gatsby to the new framework. This template is intentionally structured with user-friendly migration in mind:

The `src/content` folder contains all markdown files for pages and posts, while the `src/util` folder houses configuration files such as dark-theme-colors.json, default-colors.json, site.json, and socialmedia.json.

This organization allows users to easily copy and paste their existing content and settings into the new framework, making the transition as seamless as possible.

**Comment by munawar-js** [comment]
The core of **Foundation** revolves around markdown files. In Gatsby, we query this markdown data using GraphQL and pass the resulting props to template files located in the `src/templates` directory.

<img width="182" alt="image" src="https://github.com/user-attachments/assets/3dfadcd5-d5a3-4b5a-a2b2-25a959a8416b">

Example implementation of [index page](https://github.com/stackrole/gatsby-starter-foundation/blob/main/src/templates/index-page.js)

**Comment by munawar-js** [comment]
I’ve been weighing the options between **Next.js** and **Astro** for **Foundation**, and I’m leaning heavily towards Astro. What stands out for me is Astro’s flexibility to work with different frameworks, which is a huge plus. The built-in Markdown support is a game-changer for our content needs, and the super-fast build times make it even more appealing.

Astro’s static-first approach offers excellent scalability with very low build complexity, which means less hassle and more efficiency. Plus, the low maintenance and fast development cycles align perfectly with what we need for Foundation. Overall, Astro seems like the best fit for what we’re aiming to achieve.

**Comment by munawar-js** [comment]
Check out how simple it is to set up Markdown with Astro! The content is placed in `/src/content/pages/index.md`, and the templates are in `/src/pages/index.astro`. 

```markdown
# /src/content/pages/index.md
---
title: "Home Page"
description: "Welcome to my Astro site!"
---

# Hello, Astro!

This is a page generated from Markdown.
``` 

```astro
# /src/pages/index.astro
---
import content from '../content/pages/index.md';
---

<Layout>
  <h1>{content.frontmatter.title}</h1>
  <p>{content.frontmatter.description}</p>
  <article>
    <Markdown content={content.content} />
  </article>
</Layout>
```

This setup is super straightforward. Astro’s built-in support for Markdown and its seamless integration with templates makes the whole process incredibly efficient and clean. Just write your Markdown content, import it into your Astro template, and you’re done! No complex configurations or extra dependencies—just simple, fast, and effective development.

**Comment by munawar-js** [comment]
I made a switch, here what our tech stack looks like

<img width="537" alt="image" src="https://github.com/user-attachments/assets/417f2875-286d-4c92-b6fe-2489760f04be">

**Reply by charles-hood** [reply]
So Astro is out of the running?

**Reply by munawar-js** [reply]
Yes, we are considering it. While Astro is quite user-friendly and offers a vast ecosystem of plugins like Gatsby, it also introduces dependency on third-party updates, which can slow us down. After careful consideration, we’ve decided to go with Next.js instead. It’s a more standardized framework with an excellent developer community, and since it’s based on React, it provides everything we need out of the box without relying heavily on plugins, unlike Astro.
